### ISSUE TITLE
Carousel resources and inline code should only be included where needed

### ISSUE THREAD

**Issue by galbaras** [Automattic/jetpack#18774]
The inline code, stylesheet and scripts required for carousels are included on every page, resulting in PageSpeed issues like "unused JavaScript" and "unused CSS".

Unless the page contains a gallery, there's no need to include these.

**Comment by jeherve** [comment]
We currently only enqueue assets when needed. Could it be that there are single images on your page, that can also trigger a Carousel view? 

If you don't use single image carousels, what happens when you add the following snippet to your site?

`add_filter( 'jp_carousel_maybe_disable_single_images', '__return_true' );`

**Comment by galbaras** [comment]
The filter works. Galleries still open in carousels, but posts no longer load the carousel resources.

However, I don't see any linked images or galleries on the pages I examined, except WooCommerce product images in widgets (linked to the respective product pages) and related post thumbnails.

You're welcome to check on [here](https://dev.get-business-online.com/ronitbaras/emotional-intelligence/personal-development/talent-success-almost-made-olympic-team/).

**Comment by jeherve** [comment]
> I don't see any linked images or galleries on the pages I examined, except WooCommerce product images in widgets (linked to the respective product pages) and related post thumbnails.

Yes, we currently enqueue the assets as soon as there is an image on the post, we don't check if the image is linked. That's definitely something we should improve.

> the browser console shows this:

Yes, this is tracked in #11191.

**Comment by galbaras** [comment]
`Yes, we currently enqueue the assets as soon as there is an image on the post, we don't check if the image is linked. That's definitely something we should improve.`

Why not just make the `jp_carousel_maybe_disable_single_images` behavior `true` by default and use a SETTING to choose to add carousel code to single images? Until this issue, I didn't even know this was happening, and it was affecting the vast majority of pages on every site I have.
